src/Patterns/templates/macros.twig

```diff
@@ -0,0 +1,23 @@
+{%- macro selectFromColumns(columns, prefix = "", asLower = false) -%}
+    {%- for column in columns -%}
+        {{ prefix }}{{ column | quoteIdentifier }}
+        {%- if asLower == true %} AS {{ column | lower | quoteIdentifier }}{% endif -%}
+        {%- if not loop.last %}, {% endif -%}
+    {%- endfor -%}
+{%- endmacro -%}
+
+{%- macro generateJoin(columns, firstAlias, secondAlias) -%}
+    {%- for column in columns -%}
+        {{  firstAlias }}.{{ column | lower | quoteIdentifier }} = {{ secondAlias }}.{{ column | quoteIdentifier }}
+        {%- if not loop.last %} AND {% endif -%}
+    {%- endfor -%}
+{%- endmacro -%}
+
+{%- macro generateSnapshotPrimaryKey(snapshotPrimaryKeyParts, snapshotPrimaryKeyName) -%}
+    {%- for column in snapshotPrimaryKeyParts -%}
+        {{ column | lower | quoteIdentifier }}
+        {%- if not loop.last %} || '|' || {% endif -%}
```

Comment:
tady místo toho (nefunguje na synapse):
```
 "identifier" || '|' || "username" || '|' || "start_date" AS "snapshot_pk"
```

by bylo lepší použít 
```
concat("identifier", '|', "username", '|', "start_date") AS "snapshot_pk"
```

to by mělo fungovat i na snflk i na synapse

Reply:
synapse ma vlastnu sablonu, takze sa to tam moze spravit uplne inak (ako aj ostatne veci), ... ale kludne to opravim, `concat` je viac standardna cesta, si myslim `||` operator asi ine DB nemaju ... takze sekundu

Reply:
jj vím že bude mít, ale čím víc to bude podobný, tím to bude jednodušší na ozkoušení

Reply:
takze to bude vyzerat nejak takto, commitujem:
```
-- Merge partial results to final snapshot --
CREATE OR REPLACE TABLE "new_snapshot" AS
    SELECT
        CONCAT("id1", '|', "id2", '|', "snapshot_date") AS "snapshot_pk",
        "id1", "id2", "name", "age", "job", "snapshot_date", "actual"
    FROM "last_state"
        UNION
    SELECT
        CONCAT("id1", '|', "id2", '|', "snapshot_date") AS "snapshot_pk",
        "id1", "id2", "name", "age", "job", "snapshot_date", "actual"
    FROM "previous_state";
```

Reply:
https://github.com/keboola/app-transformation-pattern-scd/pull/14/commits/ae6c6de60301076fcf8acf7afacabb7962d0edac